Samples/resource-owner-password-credential/Angular2/openiddict-angular2-server/src/openiddict-angular2/Controllers/AccountController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.Extensions.Logging;
+using openiddict_angular2.Models;
+using openiddict_angular2.Data;
+using openiddict_angular2.Models.AccountViewModels;
+
+
+namespace openiddict_angular2.Controllers
+{
+    [Authorize]
+
+    public class AccountController : Controller
+    {
+       
+        private readonly UserManager<ApplicationUser> _userManager;
+        private readonly SignInManager<ApplicationUser> _signInManager;
+        private readonly ILogger _logger;
+
+        public AccountController(
+                                    UserManager<ApplicationUser> userManager,
+                                    SignInManager<ApplicationUser> signInManager,
+                                    ILoggerFactory loggerFactory,
+                                    ApplicationDbContext applicationDbContext
+                                 )
+            {
+                _userManager = userManager;
+                _signInManager = signInManager;
+                _logger = loggerFactory.CreateLogger<AccountController>();
+            }
+
+
+        // Custom Register
+        [Route("api/account/register")]
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> RegisterHttp([FromBody]RegisterViewModel dto)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            //currently Email and Username are same 
+            var user = new ApplicationUser { UserName = dto.Email, Email = dto.Email };
+            var result = await _userManager.CreateAsync(user, dto.Password);
```

Comment:
Missing check to ensure creation is actually succeded, see [here](http://docs.asp.net/en/latest/security/authentication/identity.html)